openedx/core/djangoapps/content_staging/filters.py

```diff
@@ -0,0 +1,55 @@
+"""
+Filters that affect the behavior of staged content (and the clipboard)
+"""
+# pylint: disable=unused-argument
+from __future__ import annotations
+
+from attrs import asdict
+
+from openedx_filters import PipelineStep
+from openedx_filters.tooling import OpenEdxPublicFilter
+from .data import StagedContentFileData
+from .models import StagedContent
+
+
+class StagingStaticAssetFilter(OpenEdxPublicFilter):
+    """
+    A filter used to determine which static assets associate with an XBlock(s)
+    should be staged in the StagedContent app (e.g. the clipboard).
+
+    This API is considered BETA. Once it is stable, this definition should be moved into openedx_filters.
+    """
+
+    filter_type = "org.openedx.content_authoring.staged_content.static_filter_source.v1"
+
+    @classmethod
+    def run_filter(cls, staged_content: StagedContent, file_datas: list[StagedContentFileData]):
+        """
+        Run this filter, which requires the following arguments:
+            staged_content (StagedContent): details of the content being staged, as saved to the DB.
+            file_datas (list[StagedContentFileData]): details of the files being staged
+        """
+        data = super().run_pipeline(staged_content=staged_content, file_datas=file_datas)
+        return data.get("file_datas")
+
+
+class IgnoreLargeFiles(PipelineStep):
+    """
+    Don't copy files over 10MB into the clipboard
+    """
+
+    # pylint: disable=arguments-differ
+    def run_filter(self, staged_content: StagedContent, file_datas: list[StagedContentFileData]):
+        """
+        Filter the list of file_datas to remove any large files
+        """
+        limit = 10 * 1024 * 1024
```

Comment:
Is there a specific set of files we're expecting to get filtered at this point?

Reply:
No, I just wanted to avoid performance issues if someone copies an XBlock that links to a huge file that's in Files & Uploads. Don't want to block the copy from completion while huge files are uploaded to temp storage that may not even be needed.

So in such a case (copying something that references a huge file), now we'll just skip that file, and intra-course copy-paste will still work for, and for cross-course copy-paste the user can copy the large asset manually.